xgboost/xgboost_cv_terminatorcallback_integration.py

```diff
@@ -0,0 +1,70 @@
+"""
+Optuna example that demonstrates a pruner for XGBoost.cv.
+
+In this example, we optimize the validation auc of cancer detection using XGBoost.
+We optimize both the choice of booster model and their hyperparameters. Throughout
+training of models, a pruner observes intermediate results and stops unpromising trials,
+and a terminator observes studies and stops optimizations with insignificant changes.
+
+You can run this example as follows:
+    $ python xgboost_cv_integration.py
+
+"""
+
+import optuna
+from optuna.terminator import report_cross_validation_scores
+from optuna.terminator import TerminatorCallback
+
+import sklearn.datasets
+import xgboost as xgb
+
+
+def objective(trial):
+    train_x, train_y = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    dtrain = xgb.DMatrix(train_x, label=train_y)
+
+    param = {
+        "verbosity": 0,
+        "objective": "binary:logistic",
+        "eval_metric": "auc",
+        "booster": trial.suggest_categorical("booster", ["gbtree", "gblinear", "dart"]),
+        "lambda": trial.suggest_float("lambda", 1e-8, 1.0, log=True),
+        "alpha": trial.suggest_float("alpha", 1e-8, 1.0, log=True),
+    }
+
+    if param["booster"] == "gbtree" or param["booster"] == "dart":
+        param["max_depth"] = trial.suggest_int("max_depth", 1, 9)
+        param["eta"] = trial.suggest_float("eta", 1e-8, 1.0, log=True)
+        param["gamma"] = trial.suggest_float("gamma", 1e-8, 1.0, log=True)
+        param["grow_policy"] = trial.suggest_categorical("grow_policy", ["depthwise", "lossguide"])
+    if param["booster"] == "dart":
+        param["sample_type"] = trial.suggest_categorical("sample_type", ["uniform", "weighted"])
+        param["normalize_type"] = trial.suggest_categorical("normalize_type", ["tree", "forest"])
+        param["rate_drop"] = trial.suggest_float("rate_drop", 1e-8, 1.0, log=True)
+        param["skip_drop"] = trial.suggest_float("skip_drop", 1e-8, 1.0, log=True)
+
+    pruning_callback = optuna.integration.XGBoostPruningCallback(trial, "test-auc")
+    history = xgb.cv(param, dtrain, num_boost_round=100, callbacks=[pruning_callback])
+
+    report_cross_validation_score(trial, history["test-auc-mean"])
+    
+    mean_auc = history["test-auc-mean"].values[-1]
```

Comment:
A nit, but please remove the trailing space, please?
https://github.com/optuna/optuna-examples/actions/runs/5126690069/jobs/9221509147?pr=194

```suggestion
    report_cross_validation_score(trial, history["test-auc-mean"])

    mean_auc = history["test-auc-mean"].values[-1]
```